homeassistant/components/switchbot_cloud/entity.py

```diff
@@ -48,3 +48,8 @@ async def send_api_command(
             command_type,
             parameters,
         )
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity is about to be added to hass."""
+        await super().async_added_to_hass()
+        self._handle_coordinator_update()
```

Comment:
Yes and no. At the end of the handle coordinator update we write state. So with this change we write state twice on creating entities. So let's create a second function that contains all the logic to set the attributes and call that from here and from the handle coordinator update

Reply:
Thank you, I'm not familiar enough with the HA codebase to realize that was happening.

PR updated with a `set_attributes` function to avoid the double writing of the state